src/Microsoft.OpenApi.Readers/ParseNodes/JsonPointerExtensions.cs

```diff
@@ -12,32 +13,32 @@ namespace Microsoft.OpenApi.Readers.ParseNodes
     public static class JsonPointerExtensions
     {
         /// <summary>
-        /// Finds the YAML node that corresponds to this JSON pointer based on the base YAML node.
+        /// Finds the JSON node that corresponds to this JSON pointer based on the base Json node.
         /// </summary>
-        public static YamlNode Find(this JsonPointer currentPointer, YamlNode baseYamlNode)
+        public static JsonNode Find(this JsonPointer currentPointer, JsonNode baseJsonNode)
         {
             if (currentPointer.Tokens.Length == 0)
             {
-                return baseYamlNode;
+                return baseJsonNode;
             }
 
             try
             {
-                var pointer = baseYamlNode;
+                var pointer = baseJsonNode;
                 foreach (var token in currentPointer.Tokens)
                 {
-                    var sequence = pointer as YamlSequenceNode;
+                    var array = pointer as JsonArray;
 
-                    if (sequence != null)
+                    if (array != null)
                     {
-                        pointer = sequence.Children[Convert.ToInt32(token)];
+                        pointer = array[Convert.ToInt32(token)];
```

Comment:
maybe use a try convert and hoist the conversion to the parent condition instead, it'll be safer

Reply:
Do you mind providing an example for this?

Reply:
assuming the token value is a string you can update the condition above to 
```CSharp
if (array != null && int32.TryParse(token, out var tokenValue))
```

and then use the new tokenValue in the indexer